Tools/AP_Periph/AP_Periph.cpp

```diff
@@ -242,6 +242,15 @@ void AP_Periph_FW::init()
     hwesc_telem.init(hal.serial(HAL_PERIPH_HWESC_SERIAL_PORT));
 #endif
 
+#ifdef HAL_PERIPH_ENABLE_ESC_APD
+    apd_esc_telem[0].init(hal.serial(APD_ESC_SERIAL_0), g.pole_count[0]);
+  #if APD_ESC_INSTANCES > 1
+    static_assert(APD_ESC_SERIAL_0 != APD_ESC_SERIAL_1, "APD ESC's must have different serial numbers");
```

Comment:
i'd like parameters for the two serial port numbers, with the defaults in hwdef.dat, like we do for other serial devices (GPS, rangefinder etc)

Reply:
Done. I had to rework to dynamically allocating the periph instance to cope with that smoothly, but it's probably more appropriate anyways.